static/js/disease/chart.ts

```diff
@@ -17,14 +17,253 @@ import * as d3 from "d3";
+  name: string;
+  // confidence score
+  score: number;
+  // lower confidence value = score - (interval/2)
+  lowerInterval: number;
+  // upper confidence value = score + (interval/2)
+  upperInterval: number;
+}
+/**
+ * Draws the disease-gene association charts for the disease of interest
+ * @param id
+ * @param data
+ * @returns
+ */
 export function drawDiseaseGeneAssocChart(
   id: string,
-  data: DiseaseGeneAssociationData[]
+  data: DiseaseGeneAssociationRawData[]
 ): void {
-  return;
+  // checks if the data is empty or not
+  if (_.isEmpty(data)) {
+    return;
+  }
+  let reformattedData = [] as DiseaseGeneAssociationIntervalData[];
+  // reformats the data by converting the confidence interval length to lower and upper confidence values
+  reformattedData = data.map((item) => {
+    return {
+      name: item.name,
+      score: Number(item.score),
+      lowerInterval: Number(item.score - item.interval / 2),
+      upperInterval: Number(item.score) + Number(item.interval / 2),
+    };
+  });
+  //Formats the gene name
+  function formatGeneName(d: string) {
+    // remove the word "bio/hg38_" from the entire gene name, say "bio/hg38_pRNA"
+    d = d.substring(9);
+    return d;
+  }
+  const height = GRAPH_HEIGHT_M - MARGIN.top - MARGIN.bottom;
+  const width = GRAPH_WIDTH_L - MARGIN.left - MARGIN.right;
+  const svg = d3
+    .select("#disease-gene-association-chart")
+    .append("svg")
+    .attr("width", width + MARGIN.left + MARGIN.right)
+    .attr("height", height + MARGIN.top + MARGIN.bottom)
+    .append("g")
+    .attr("transform", "translate(" + MARGIN.left + "," + MARGIN.top + ")");
+  // slicing the data to display 10 values only
+  reformattedData = reformattedData.slice(0, NUM_DATA_POINTS);
+  // plots the axes
+  const x = d3
+    .scaleBand()
+    .range([0, width])
+    .domain(
+      reformattedData.map(function (d) {
+        return d.name;
+      })
+    )
+    .padding(1);
+  svg
+    .append("g")
+    .attr("transform", "translate(0," + height + ")")
+    .call(d3.axisBottom(x).tickFormat(formatGeneName))
+    .selectAll("text")
+    .attr("transform", "translate(-10,0)rotate(-45)")
+    .style("text-anchor", "end");
+  addXLabel(width, height, "Gene Names", svg);
+  const y = d3
+    .scaleLinear()
+    .domain([0, d3.max(reformattedData, (d) => d.upperInterval) + Y_AXIS_LIMIT])
+    .range([height, 0]);
+  svg.append("g").call(d3.axisLeft(y));
+  addYLabel(height, "Confidence Score", svg);
+  // the lines
+  svg
+    .selectAll("myline")
+    .data(reformattedData)
+    .enter()
+    .append("line")
+    .attr("x1", (d) => {
+      return x(d.name);
+    })
+    .attr("x2", (d) => {
+      return x(d.name);
+    })
+    .attr("y1", (d) => {
+      return y(d.lowerInterval);
+    })
+    .attr("y2", (d) => {
+      return y(d.upperInterval);
+    })
+    .attr("stroke", "grey");
+  const circleIDFunc = getElementIDFunc(id, "circle");
+  // the circles
+  svg
+    .selectAll("mycircle")
```

Comment:
we don't want to use possessive adjectives (like "my") in selector names, this can just be "line" or "disease-gene-line" to be more specific